docs/book/cookbook/autowiring-routes-and-pipelines.md

```diff
@@ -1,210 +1,8 @@
-- Ensure you only define the pipeline in a **single** delegator factory on your
-  `Application` instance.
-
-## Caution: third-party, distributed modules
-
-If you are developing a module to distribute as a package via
-[Composer](https://getcomposer.org/), **you should not autowire any delegator
-factories that inject pipeline middleware or routes in the `Application`**.
-
-Why?
-
-As noted in the above section, pipelines should be created exactly once, at
-the application level. Registering pipeline middleware within a distributable
-package will very likely not have the intended consequences.
-
-If you ship with pipeline middleware, we suggest that you:
-
-- Document the middleware, and where you anticipate it being used in the
-  middleware pipeline.
-- Document how to add the middleware service to dependency configuration, or
-  provide the dependency configuration via your module's `ConfigProvider`.
-
-With regards to routes, there are other considerations:
-
-- Routes defined by the package might conflict with the application, or with
-  other packages used by the application.
-
-- Routing definitions are typically highly specific to the router implementation
-  in use. As an example, each of the currently supported router implementations
-  has a different syntax for placeholders:
-
-    - `/user/:id` + "constraints" configuration to define constraints (zend-router)
-    - `/user/{id}` + "tokens" configuration to define constraints (Aura.Router)
-    - `/user/{id:\d+}` (FastRoute)
-
-- Your application may have specific routing considerations or design.
-
-You could, of course, detect what router is in use, and provide routing for each
-known, supported router implementation within your delegator factory. We even
-recommend doing exactly that. However, we note that such an approach does not
-solve the other two points above.
-
-However, we still recommend _shipping_ a delegator factory that would register
-your routes, since routes *are* often a part of module design; just **do not
-autowire** that delegator factory. This way, end-users who *can* use the
-defaults do not need to cut-and-paste routing definitions from your
-documentation into their own applications; they will instead opt-in to your
-delegator factory by wiring it into their own configuration.
-
-## Synopsis
-
-- We recommend using delegator factories for the purpose of autowiring routes,
-  and, with caveats, pipeline middleware:
-    - The pipeline should be created exactly once, so calls to `pipe()` should
-      occur in exactly _one_ delegator factory.
-- Distributable packages should create a delegator factory for _routes only_,
-  but _should not_ register the delegator factory by default.
+<noscript><meta http-equiv="refresh" content="0; url=v2/cookbook/autowiring-routes-and-pipelines/"></noscript>
+<script>
+  document.addEventListener("DOMContentLoaded", function (event) {
+    var uri = new URL(window.location.href);
```

Comment:
The problem still exists: it will not work in IE.

You can simplify the script:

```javascript
document.addEventListener('DOMContentLoaded', function (event) {
    window.location.pathname = 'v2/cookbook/autowiring-routes-and-pipelines/';
});
```

Reply:
In which versions of IE will it not work?

Based on our metrics, IE users are a sliver of a percent of users; most are on (in decreasing order) Chrome, Firefox, Safari, or Edge. The majority of IE users are on 11. If the syntax will work on IE 11, the above will work.

Additionally, window.location.pathname is _read-only_. We would need to use `location.replace`, which expects a fully qualified URI... which the proposed solution accomplishes using the URL class.

Reply:
> In which versions of IE will it not work?

All:

* https://caniuse.com/#feat=url
* https://developer.mozilla.org/en-US/docs/Web/API/URL#Browser_compatibility

Reply:
> Additionally, window.location.pathname is read-only.

But works.

> We would need to use `location.replace`…

Which is correct for HTTP redirects.

```javascript
document.location.replace(
    document.location.protocol + '//' + document.location.host +  '/v2/cookbook/autowiring-routes-and-pipelines/'
);
```